### PR TITLE
Intensity adjustment. Find a better upper bound for interpolating images. 

### PR DESCRIPTION
**Pull request by Tingyiwanyan**
adding intensity_adjustment function

### CODE REVIEW

File: dipy/viz/tests/test_visualization.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import numpy as np
```
Comment (riddhishb) [inline dipy/viz/tests/test_visualization.py]: @Tingyiwanyan If this a new functionality you are proposing which I think it is, then the function should go in the folder data/viz/ and not the tests folder. The data/viz/tests folder is for providing the functional tests using nosetests which you will have to do for your proposed function as well. Secondly, the tutorial you have provided should be renamed to some other name (check other examples). If this function is a work in progress then add the WIP tag to your PR. :)

**Comment by riddhishb** [comment]
@Tingyiwanyan Check my comment on the code which I have made.

**Comment by Garyfallidis** [comment]
Hi @Tingyiwanyan you have left a figure function and it the test cannot pass https://travis-ci.org/nipy/dipy/jobs/162845994#L2462
Thanks.

File: dipy/viz/tests/test_visualization.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    r""" This function enables changing the MRI image contrast.
+
+    It first read the data, must be a 2D image (a specific slice image in MRI),
+    then calaulate the image intensity histogram, and based on the rate value to
+    decide what is the upperbound value for intensity normalization, the lowerbound
+    is 0.
+
+    Parameters
+    -----------
+    data : a two dimensional array representing one MRI slice image
```
Comment (Garyfallidis) [inline dipy/viz/tests/test_visualization.py]:
```
data : ndarray
```

**Comment by Garyfallidis** [comment]
When I am calculating the coverage of which is the only (no test file added here). I get that everything new in this function has been already tested. There are no missing lines for any new code. 

dipy.segment.threshold         41     10    76%   22-40
22-40 are older lines which I think are tested from a different place. However the message in coverage/coveralls says that the coverage was decreased by 0.1% 

I am not sure why coverage looks like is reducing with this PR? Any ideas?

**Comment by Garyfallidis** [comment]
And here is what the Travis bots with coverage is reporting
dipy/segment/threshold.py                             41      0     16      1    98%
https://travis-ci.org/nipy/dipy/jobs/167766465
But looking at another PR which branched earlier I see that the coverage was 100%
dipy/segment/threshold.py                             12      0      0      0   100%

So, should we be trusting coveralls with small differences?

**Comment by arokem** [comment]
Yeah - I have not found coveralls to be informative, and have generally
been ignoring those messages :-(

**Comment by Garyfallidis** [comment]
Yeah, okay. Will merge this one then. Congrats @Tingyiwanyan ! This is your first PR in DIPY :+1:

**Comment by Garyfallidis** [comment]
And @arokem we should rethink about coveralls. We are already calculating the coverage in different bots. 
Also look at the badge also in the README file. It says unknown. 
https://github.com/nipy/dipy
Should we at least remove this one? If not used?

File: dipy/segment/threshold.py [hunk 3]
```diff
@@ -38,3 +38,77 @@ def otsu(image, nbins=256):
...
+    g, h = np.histogram(data)
+    m = np.zeros((10, 3))
+    low = data.min()
+    high = data.max()
+    for i in np.array(range(10)):
+        m[i, 0] = g[i]
+        m[i, 1] = h[i]
+        m[i, 2] = h[i + 1]
+
+    g = sorted(g,reverse = True)
```
Comment (arokem) [inline dipy/segment/threshold.py]: PEP8: whitespace after comma

File: dipy/segment/threshold.py [hunk 4]
```diff
@@ -38,3 +38,77 @@ def otsu(image, nbins=256):
...
+    sz = np.size(g)
+
+    Index = 0
```
Comment (arokem) [inline dipy/segment/threshold.py]: PEP8: variable names should begin with a lower-case letter

**Comment by arokem** [comment]
@Garyfallidis : would you please give advance warning before you go ahead and merge PRs?

**Comment by Garyfallidis** [comment]
Hey, we said that for tiny PRs there is no reason to wait amongst the core devs. Any core dev can merge those.

**Comment by arokem** [comment]
I don't see the need to hurry to merge things (except when there is some
particular need to hurry). I would still prefer a heads-up. At the very
least, we would have avoided those PEP8 violations in master.

**Comment by jchoude** [comment]
I don't like reviving merged PRs, but going through my emails and seeing this, I was curious on what is included. Just looking at the code quickly, I see a few things (like the PEP8 comments from @arokem , and some documentation comments). Should I still comment, or just open issues?

**Comment by jchoude** [comment]
Also, sorry for the delay, getting back on top of the stack now.

**Comment by arokem** [comment]
Good to open issues on these -- even better to open PRs ;)

**Pull request merged**